components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -108,8 +109,7 @@ export function CreateWorkspacePage() {
                 workspaceAutostartOptions: workspaceAutoStartOptions,
             });
         }
-        setUser(user);
-        getGitpodService().server.updateLoggedInUser(user).catch(console.error);
+        await getGitpodService().server.updateLoggedInUser(user).then(setUser).catch(console.error);
```

Comment:
setting the user first and not waiting for the server response would create strange state when a redirect happens next. I.e. the updateLoggedIn could possibly never execute.